public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.tsx

```diff
@@ -1,23 +1,43 @@
 import { PanelChrome } from '@grafana/ui';
 import { PanelRenderer } from 'app/features/panel/PanelRenderer';
-import React, { useState } from 'react';
-import { PanelModel } from '../../state';
+import React, { useCallback, useEffect, useState } from 'react';
+import { PanelModel, DashboardModel } from '../../state';
 import { usePanelLatestData } from './usePanelLatestData';
 import { PanelOptions } from 'app/plugins/panel/table/models.gen';
-
+import { RefreshEvent } from 'app/types/events';
+import { Subscription } from 'rxjs';
+import { applyPanelTimeOverrides } from 'app/features/dashboard/utils/panel';
+import { getTimeSrv, TimeSrv } from '../../services/TimeSrv';
 interface Props {
   width: number;
   height: number;
   panel: PanelModel;
+  dashboard: DashboardModel;
 }
 
-export function PanelEditorTableView({ width, height, panel }: Props) {
-  const { data } = usePanelLatestData(panel, { withTransforms: true, withFieldConfig: false }, true);
+export function PanelEditorTableView({ width, height, panel, dashboard }: Props) {
+  const { data } = usePanelLatestData(panel, { withTransforms: true, withFieldConfig: false }, false);
   const [options, setOptions] = useState<PanelOptions>({
     frameIndex: 0,
     showHeader: true,
   });
 
+  const timeSrv: TimeSrv = getTimeSrv();
+  const timeData = applyPanelTimeOverrides(panel, timeSrv.timeRange());
+
+  const onRefresh = useCallback(() => {
+    panel.runAllPanelQueries(dashboard.id, dashboard.getTimezone(), timeData, width);
+  }, [dashboard, panel, timeData, width]);
+
+  // Subscribe to panel events
+  useEffect(() => {
+    const subs = new Subscription();
```

Comment:
Not sure how `subs` will be treated when this is out of scope? Do you know? Maybe we should create a proper ref instead? See https://github.com/grafana/grafana/blob/fcdfa880899a0b7080234166da97a2eec0f5868d/public/app/features/dashboard/components/PanelEditor/usePanelLatestData.ts#L22

Reply:
why? this is not out of scope anywhere? this is how we do it everywhere else we create subscriptions inside a useEffect. Never used a ref for it, not sure why we would need to, it's not used outside the useEffect. 

But we not need to use Subscription here as we only have one Unsubscribable, so we can just do: 

```ts
useEffect(() => {

    const sub = panel.events.subscribe(RefreshEvent, () => {
        panel.runAllPanelQueries(dashboard.id, dashboard.getTimezone(), timeData, width);
    }));

    return () => sub.unsubscribe;
}, [onRefresh, panel, dashboard, timeData, width]);
```

Reply:
You're right Torkel, I was reading it wrong.